src/persist-client/src/read.rs

```diff
@@ -400,25 +377,17 @@ where
     /// TODO: If/when persist learns about the structure of the keys and values
     /// being stored, this is an opportunity to push down projection and key
     /// filter information.
-    pub async fn listen(
-        &self,
-        timeout: Duration,
-        as_of: Antichain<T>,
-    ) -> Result<Result<Listen<K, V, T, D>, InvalidUsage>, ExternalError> {
-        trace!("ReadHandle::listen timeout={:?} as_of={:?}", timeout, as_of);
+    pub async fn listen(&self, as_of: Antichain<T>) -> Result<Listen<K, V, T, D>, Since<T>> {
```

Comment:
I don't think I quite follow exactly why `Since<T>` here (and similar places) isn't part of the `InvalidUsage` enum?  I'm not saying I think it's _wrong_ because this is analogous to `compare_and_append` -- but is the general rule that if `Err(<Antchain>)`  had been the applicable argument, the call would have succeeded?  If yeah, maybe worth adding a note to explain on this function and others

Reply:
Since is pretty categorically different from InvalidUsage. it was previously lumped in, but this is a change that came out of aljoscha and I using the persist api ourselves: InvalidUsage was mostly something you should panic on as a user.. except this, which might be interesting for the user to act on in a structured way. this is somewhat buried atm in one of the commit messages in this PR, but the New Way (tm) is that persist users should likely just always panic in response to InvalidUsage, it means "you done messed up" whereas the Since and Upper responses are simply indications that you have out of date information. hopefully that makes sense!

adding to the docs definitely sgtm though! that's a great point

Reply:
okay yeah this matches my intuition -- thanks for the explanation!